moto/sqs/models.py

```diff
@@ -518,10 +518,8 @@ def add_message(self, message):
             if result:
                 [backend.delete_message(self.name, m.receipt_handle) for m in messages]
             else:
-                [
-                    backend.change_message_visibility(self.name, m.receipt_handle, 0)
-                    for m in messages
-                ]
+                # Make messages visible again
```

Comment:
Logic improvement, some tests failed here because of a race condition.
The old logic assumed the message was invisible, and fail hard if it wasn't.